### PR TITLE
feat(label): add support for dropdown label

### LINKED ISSUE #26: auro-dropdown: add a label to the dropdown
## Is your feature request related to a problem? Please describe.

Auro-dropdown needs to support a label per the figma design.

## Describe the solution you'd like

Label should match the [figma design](https://www.figma.com/file/3TQhiKjQqQOKYqaqcO1zj2/Dropdown?node-id=906%3A753).

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #26

## Summary:

This change adds support for a label on `auro-dropdown`.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/auro-dropdown.js [hunk 1]
```diff
@@ -200,8 +201,13 @@ class AuroDropdown extends LitElement {
           <slot role="tooltip"></slot>
         </div>
         <div id="trigger" data-trigger-placement="${this.placement}">
-          <div id="triggerContent" chevron=${this.chevron}>
-            <slot name="trigger"></slot>
+          <div>
```
Comment (blackfalcon) [inline src/auro-dropdown.js]: Wouldn't be easier to manage if this was a slot versus a property? Like https://github.com/AlaskaAirlines/auro-input/blob/main/src/auro-input.js#L294-L299

File: src/style.scss [hunk 2]
```diff
@@ -61,6 +61,12 @@
   }
 }
 
+#label {
```
Comment (blackfalcon) [inline src/style.scss]: I realize there are some poor practices with this CSS, but maybe curb the madness and make this a class instead?

File: src/auro-dropdown.js [hunk 3]
```diff
@@ -49,7 +49,8 @@ class AuroDropdown extends LitElement {
       for:           { type: String },
       toggle:        { type: Boolean },
       chevron:       { type: Boolean },
-      dropdownWidth: { type: Number }
+      dropdownWidth: { type: Number },
+      label:         { type: String }
```
Comment (blackfalcon) [inline src/auro-dropdown.js]: If you make this a slot then this is not needed.

File: src/auro-dropdown.js (hunk 1 above)
Comment (blackfalcon) [reply src/auro-dropdown.js]: I am not seeing a test to support this property or if statement? Property or slot, there needs to be an addition to the JSDoc header.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: src/auro-dropdown.js (hunk 1 above)
Comment (jason-capsule42) [reply src/auro-dropdown.js]: > Wouldn't be easier to manage if this was a slot versus a property? Like https://github.com/AlaskaAirlines/auro-input/blob/main/src/auro-input.js#L294-L299

No, making this a slot opens the door for someone to put any HTML they want into the slot. There should never be anything other than a simple text string entered and using an attribute enforces that.

File: src/style.scss (hunk 2 above)
Comment (jason-capsule42) [reply src/style.scss]: There is only ever going to be one #label inside the shadow DOM container. Why would we want to use a class here?

File: src/auro-dropdown.js (hunk 3 above)
Comment (jason-capsule42) [reply src/auro-dropdown.js]: Same comment as above - slot is to open ended. There should only ever be a simple string defined here. Using the attribute enforces that where a slot makes it possible to add anything they want to a label which we do not support.

File: src/auro-dropdown.js (hunk 1 above)
Comment (blackfalcon) [reply src/auro-dropdown.js]: That's not necessarily true. The discussion around a users ability to add in things like auro-icon https://github.com/AlaskaAirlines/auro-input/issues/59

File: src/auro-dropdown.js (hunk 3 above)
Comment (blackfalcon) [reply src/auro-dropdown.js]: Same reply https://github.com/AlaskaAirlines/auro-input/issues/59

File: src/style.scss (hunk 2 above)
Comment (blackfalcon) [reply src/style.scss]: As a general rule in CSS, ID selectors are never preferred. We are in the scope of an encapsulated UI, the specificity of an ID is not necessary. 

I am not 100% as to the state of this repo, but I am also pretty sure this is in violation with our stylelint rules https://github.com/stylelint/stylelint/blob/main/lib/rules/selector-id-pattern/README.md#options

from https://github.com/stylelint/stylelint-config-standard/blob/main/index.js#L138 that is extended from stylelint-config-standard-scss

File: test/auro-dropdown.test.js [hunk 4]
```diff
@@ -1,27 +1,75 @@
...
 
-// describe('aurolabs-dropdown', () => {
-//   it('sets the CSS class on aurolabs-dropdown > div element', async () => {
-//     const el = await fixture(html`
-//       <aurolabs-dropdown cssclass="testClass"></aurolabs-dropdown>
-//     `);
+describe('auro-dropdown', () => {
+  it('auro-dropdown is accessible', async () => {
+    const el = await fixture(html`
+      <auro-dropdown cssClass="testClass"></auro-dropdown>
```
Comment (blackfalcon) [inline test/auro-dropdown.test.js]: This isn't exactly a valid test. There is no `cssClass` attribute in the API.

File: test/auro-dropdown.test.js [hunk 5]
```diff
@@ -1,27 +1,75 @@
...
+    trigger.click();
+    expectPopoverShown(el);
+    expect(chevron).to.have.attribute('data-expanded');
+
+    trigger.click();
+    expectPopoverHidden(el);
+    expect(chevron).to.not.have.attribute('data-expanded');
+  })
+
+  it('auro-dropdown programatically show and hide', async () => {
```
Comment (blackfalcon) [inline test/auro-dropdown.test.js]:
```suggestion
  it('auro-dropdown programmatically show and hide', async () => {
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]
I also noticed that the HTML template has a wrapping div that I am not sure is needed? 

```html
<div id="defaultExample">
```

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**